hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1091,13 +1102,23 @@ public void update(HoodieCommitMetadata commitMetadata, String instantTime) {
               enabledPartitionTypes, dataWriteConfig.getBloomFilterType(),
               dataWriteConfig.getBloomIndexParallelism(), dataWriteConfig.getWritesFileIdEncoding());
 
+      Option<HoodiePairData<String, List<List<HoodieColumnRangeMetadata<Comparable>>>>> partitionRangeMetadataPartitionPairOpt = Option.empty();
+      if (enabledPartitionTypes.contains(MetadataPartitionType.PARTITION_STATS)) {
+        checkState(MetadataPartitionType.COLUMN_STATS.isMetadataPartitionAvailable(dataMetaClient),
+            "Column stats partition must be enabled to generate partition stats. Please enable: " + HoodieMetadataConfig.ENABLE_METADATA_INDEX_COLUMN_STATS.key());
+        partitionRangeMetadataPartitionPairOpt = Option.of(convertMetadataToPartitionStatsColumnRangeMetadata(commitMetadata, engineContext, dataMetaClient, dataWriteConfig.getMetadataConfig()));
```

Comment:
So, this is getting called only if `enabledPartitionTypes.contains(MetadataPartitionType.PARTITION_STATS)`. In this case, can we reuse the output of `convertMetadataToPartitionStatsColumnRangeMetadata` if we ensure that records for partition stats are computed before expression index?

Reply:
This change was from the older approach. I have done removed this change now.